packages/slate-editor/src/modules/editor-v4-rich-formatting-menu/RichFormattingMenu.tsx

```diff
@@ -138,12 +139,21 @@ export const RichFormattingMenu: FunctionComponent<Props> = ({
         Transforms.select(editor, selection);
     }
 
+    React.useEffect(() => {
+        Editor.addMark(editor, MarkType.SELECTION, true);
+
+        return () => {
+            Editor.removeMark(editor, MarkType.SELECTION);
+        };
+    }, [linkRange?.current]);
```

Comment:
Applying selection marks to the document state is not the best idea. It can produce leftovers in the document value object with the marks not supported by the backend. I had this case locally after clicking link button and then dismissing the link menu after that. Selection marks were kept in the document even after I selection something else.
I believe I've fixed it in https://github.com/prezly/slate/pull/124 by refactoring selection rendering to using decorate functionality.